apis/s3/src/main/java/org/jclouds/s3/blobstore/S3BlobStore.java

```diff
@@ -174,7 +189,12 @@ public void setContainerAccess(String container, ContainerAccess access) {
    @Override
    public PageSet<? extends StorageMetadata> list(String container, ListContainerOptions options) {
       ListBucketOptions httpOptions = container2BucketListOptions.apply(options);
-      PageSet<? extends StorageMetadata> list = bucket2ResourceList.apply(sync.listBucket(container, httpOptions));
+      PageSet<? extends StorageMetadata> list = null;
+      if (httpOptions.getVersions()){
```

Comment:
This can throw `NullPointerException` when it return `null`.  Can you change the implementation of `getVersions` to return `false` when it is not specified?